libfwupdplugin/fu-smbios.c

```diff
@@ -567,8 +709,18 @@ fu_smbios_setup (FuSmbios *self, GError **error)
 
 	/* DMI */
 	path = g_build_filename (sysfsfwdir, "dmi", "tables", NULL);
-	if (g_file_test (path, G_FILE_TEST_EXISTS))
-		return fu_smbios_setup_from_path (self, path, error);
+	if (g_file_test (path, G_FILE_TEST_EXISTS)) {
+		if (fu_smbios_setup_from_path (self, path, error))
```

Comment:
I think you want to actually use a local error here.

Reply:
I'd prefer if we could chain errors, but added a local error to prefix the original one and indicate what was tried.